### ISSUE TITLE
[FEATURE]  Make optional the quick start menu

### ISSUE THREAD

**Issue by O35dE** [strongbox-password-safe/Strongbox#776]
**Preliminaries**
When you long press the application icon the quick start option menu appears along with the names of the first four DBs.

**Versions**

**On iOS (please complete the following information):**
 - Device: iPhone 15
 - OS: iOS 17.4.1

** Strongbox Version **
 - Version: 1.59.9

**Describe the bug**
Would it be possible to make this quick start menu optional? Some users may use in the name of the DB some information that they would not like to make visible from outside the application by anyone who inadvertently has access to their iPhone.

**To Reproduce**
Long press the app icon and the names of four DBs are displayed.

**Expected behavior**
Nothing is shown outside the app.

**Screenshots & Recordings**

**Additional context**

**Comment by strongbox-mark** [comment]
Yes, not a bad idea, we'll try to fit it into the next release.

**Comment by O35dE** [comment]
Thanks a lot Mark, your solution is splendid as it combines the best (by far) pw app + best support.

**Comment by strongbox-mark** [comment]
Yes, I see the issue myself too, but it actually appears to be some kind of iOS level issue. Apple must keep some kind of cache around which isn't properly cleared. I'll raise this issue with Apple via their Feedback Assistant but unfortunately these kinds of things usually just go into a blackhole...

To workaround this, the best thing you can do is remove Strongbox completely (making sure you've got copies of your databases somewhere safe). Then reinstall, go straight to preferences and turn the option OFF, so that the database names themselves never enter any Apple iOS level cache. Then you can start re-adding your databases.

**Comment by strongbox-mark** [comment]
Thanks @O35dE - When you say:

> one of those apps that lock the apps you choose by using Face Id

What are these?

The cache appears to remember the shortcut items after a restart only if they've ever been set. So, the (unfortunate) solution is to do a clean install of Strongbox, launch and go straight to settings to switch this off _before_ you add a database. Once the app never puts these shortcut items in the menu, the system never caches them. 

> Could you please test if after some device restarts the menu will remain disabled? 

Sure, we've done this, but this is also something you can do yourself too, right?

> And if it doesn’t work, I wonder if maybe it would if the default configuration for this option was “disabled”, at least in the Zero version? 

Sure, it's something we'll think about.

> Depending on the complexity of the user's data configuration, it may be necessary to have informations in the name of the DBs that should be inaccessible unless after the unlocking of the app by the authorized user.

Can you provide an example, it'll be helpful for our consideration.

**Comment by strongbox-mark** [comment]
Hi @O35dE - We've made the default OFF in the Zero build now. Could you try removing Strongbox Zero completely, then reinstall and verify that the menu no longer shows your database names?

**Comment by strongbox-mark** [comment]
OK, thanks @O35dE - back to the drawing board.

**Comment by dan-el** [comment]
**Here are my test results:**
1) "Show Databases on App Shortcuts Menu" was on. I never changed it so far.
2) Switched it to "off" and immediately after the db disappeared from the long-press menu.
3) Downloaded and installed iOS 18.0.1 (i have an iPhone 12 mini btw)
4) After the reboot I directly long-pressed the Strongbox app icon and the db name **was back in the list.**
5) I went to the app settings to confirm "Show Databases on App Shortcuts Menu" was still toggled off.
6) After the settings check (aka. launching Strongbox) the db name is gone again from the long-press menu as it should be.

Appears something is still not working as expcted.
